drivers/generic_gpio_common.h

```diff
@@ -0,0 +1,48 @@
+#ifndef GENERIC_GPIO_COMMON_H
+#define GENERIC_GPIO_COMMON_H
+
+#include <stdlib.h>
+#include <gpiod.h>
+#include <regex.h>
+#include <ctype.h>
+
+/*  rules commands definition    */
+#define RULES_CMD_NOT   -2
+#define RULES_CMD_AND   -3
+#define RULES_CMD_OR    -4
+#define RULES_CMD_OBR   -5
+#define RULES_CMD_CBR   -6
+#define RULES_CMD_LAST  RULES_CMD_CBR
+
+/*  run option definitions      */
+#define ROPT_REQRES 0x00000001  /* reserve GPIO lines only during request processing    */
+#define ROPT_EVMODE 0x00000002  /* event driven run                                     */
+
+typedef struct rulesint_t { /* structure to store processed rules configuration per each state */
+    char    stateName[12];  /* NUT state name for rules in cRules */
+    int     archVal;        /* previous state value */
```

Comment:
Just in case: does this deal with architecture-dependent sizes of `int` or some wire-protocol specific `(u)int32_t` etc.? The `nut_stdint.h` takes care of providing those for different platforms.